src/main/java/org/folio/uk/service/UserService.java

```diff
@@ -98,21 +106,55 @@ public Optional<User> getUser(UUID id) {
     return usersClient.lookupUserById(id);
   }
 
-  public CompositeUser getUserBySelfReference(List<IncludedField> include, boolean expandPermissions) {
+  public CompositeUser getUserBySelfReference(List<IncludedField> include, boolean expandPermissions,
+                                              boolean overrideUser) {
     log.info("Retrieving user by self reference with parameters: include = {}, expandPermissions = {}",
       () -> StringUtils.join(emptyIfNull(include), ", "), () -> expandPermissions);
 
     var userId = getUserId();
-
     var user = usersClient.lookupUserById(userId)
       .orElseThrow(() -> new EntityNotFoundException("User was Not Found with: id = " + userId));
 
-    return new CompositeUser()
-      .user(user)
+    // Whether to override from shadow to a real user in ECS
```

Comment:
more informative comment or something like that:
The overrideUser is set to true during login to support ECS login into member tenants.